terraform/hydrocron-dynamo.tf

```diff
@@ -27,4 +37,16 @@ resource "aws_dynamodb_table" "hydrocron-swot-node-table" {
     name = "range_start_time"
     type = "S"
   }
+  attribute {
+    name = "granuleUR"
+    type = "S"
+  }
+
+  global_secondary_index {
```

Comment:
What are the implications of adding a GSI?

A brief scan of the AWS docs a few things jump out to me:
> When you put or delete items in a table, the global secondary indexes on that table are updated in an eventually consistent fashion. Changes to the table data are propagated to the global secondary indexes within a fraction of a second, under normal conditions. However, in some unlikely failure scenarios, longer propagation delays might occur.

Good to keep in mind but I don't think this is a showstopper for our particular application

> Your AWS account is charged for storage of the item in the base table and also for storage of attributes in any global secondary indexes on that table.
> ...
> To estimate the storage requirements for a global secondary index, you can estimate the average size of an item in the index and then multiply by the number of items in the base table that have the global secondary index key attributes.

We will need to update cost models for this GSI @jimmie 

Any other considerations you found that might be important to note @torimcd ?

Reply:
One other thing I'll mention is that there are recommendations if using provisioned throughput (https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GSI.html#GSI.ThroughputConsiderations).

We are using on-demand tables so this is not applicable right now, but if the topic of switching to provisioned mode ever comes up this would need to be taken into account.